File: node.js/src/slimmed/domain/books/dtos/bookDetails.ts
export type BookDetails = {
  id: string;
  currentState: string;
  title: string;
  author: AuthorDetails;
  publisherName: string;
  edition: number;
  genre?: string | null;
  isbn?: string | null;
  publicationDate?: string | null;
  totalPages?: number | null;
  numberOfIllustrations?: number | null;
  bindingType?: string | null;
  summary?: string | null;
  committeeApproval: CommitteeApprovalDetails | null;
  reviewers: string[];
  chapters: ChapterDetails[];
  translations: TranslationDetails[];
  formats: FormatDetails[];
};

export type AuthorDetails = { firstName: string; lastName: string };
export type CommitteeApprovalDetails = {
  isApproved: boolean;
  feedback: string;
};
export type ChapterDetails = { title: string; content: string };
export type TranslationDetails = { language: string; translator: string };
export type FormatDetails = {
  formatType: string;
  totalCopies: number;
  soldCopies: number;
};

File: java/src/main/java/io/eventdriven/slimdownaggregates/original/domain/books/dtos/BookDetails.java
package io.eventdriven.slimdownaggregates.original.domain.books.dtos;

import java.time.LocalDate;
import java.util.UUID;

public record BookDetails(
  UUID id,
  String currentState,
  String title,
  AuthorDetails author,
  String publisherName,
  int edition,
  String genre,
  String isbn,
  LocalDate publicationDate,
  Integer totalPages,
  Integer numberOfIllustrations,
  String bindingType,
  String summary,
  CommitteeApprovalDetails committeeApproval,
  String[] reviewers,
  ChapterDetails[] chapters,
  TranslationDetails[] translations,
  FormatDetails[] formats

) {

  public record AuthorDetails(String firstName, String lastName) {
  }

  public record CommitteeApprovalDetails(boolean isApproved, String feedback) {
  }

  public record ChapterDetails(String title, String content) {
  }

  public record TranslationDetails(String language, String translator) {
  }

  public record FormatDetails(String formatType, int totalCopies,
                              int soldCopies) {
  }
}
